Operator 18.0.0 release notes (#1475)
keycloak/keycloak-documentation: release_notes/topics/18_0_0.adoc

= Highlights

== New Operator preview

> With this release, we're introducing a brand new {project_operator} as a tech-preview. Apart from being rewritten from

Reviewer: ```suggestion
With this release, we're introducing a brand new {project_operator} as a preview. Apart from being rewritten from
```

We don't use tech-preview in community, but rather just preview

scratch, the main user-facing change from the legacy Operator is the used {project_name} distribution – the new Operator

> uses the Quarkus distribution of {project_name}. With that, there are a few breaking changes. For details, incl.

Reviewer 2: Since we changed the API group "breaking changes" is not technically correct, what do you think about something along those lines?
```suggestion
uses the Quarkus distribution of {project_name}. With that, the API (in form of Custom Resource Definitions) has changed. For details, incl.
```

> installation and migration instructions, see the https://www.keycloak.org/guides[Operator related guides].

Reviewer 3: ```suggestion
installation and migration instructions, see the https://www.keycloak.org/guides#operator[Operator related guides].
```

Reviewer 3: Adding the fragment scrolls directly to the section, see https://www.keycloak.org/guides#server

> The link:{operatorRepo_link}[legacy Operator] will receive updates until June 2022 when the {project_name} WildFly

Reviewer 3: Is June actually included? Then I'd write until start of July or sth. instead until is a bit ambiguous at times ;)

Author: @DGuhr Thank you for your review!

I'm simply aligning it with https://github.com/keycloak/keycloak-documentation/blob/main/release_notes/topics/17_0_0.adoc#quarkus-distribution-is-now-fully-supported. :)
> With that in mind we will continue to support the WildFly distribution until June 2022.

> The link:{operatorRepo_link}[legacy Operator] will receive updates until June 2022 when the {project_name} WildFly

Reviewer: ```suggestion
The link:{operatorRepo_link}[legacy Operator] will receive updates until Keycloak 20 when the {project_name} WildFly
```

See https://www.keycloak.org/2022/03/releases.html

distribution reaches EOL.

=== OperatorHub versioning scheme
To avoid version conflicts with the legacy Operator, the 18.0.0 version of the new Operator is released as version

> `20.0.0-alpha.1` on OperatorHub. The legacy Operator versioning scheme remains the same, i.e. it is released as 18.0.0.

Reviewer 3: Just a sidenote: This _could_ slightly impact the overall usage/adoption of the new operator btw. I know many companies where compliance does not allow you to  install anything tagged _alpha_, _beta_ or sth. And I know from myself I either delay the usage than filing multiple compliance reports to get the use of a "production ready operator thats only tagged wrong for technical reasons" through the hells of bureaucracy. Can't come up with a better solution myself, though. ;)

Author: That's a good point, but I think we don't want to hide too much the fact that it's a tech-preview. After all, the new operator will live in `candidate` channel for now.

Reviewer 2: I have to add that versioning stuff with `alpha` and `beta` is a pretty common practice on Kubernetes land.

> The same pattern will apply for future {project_name} 18 and 19 releases, until version 20 where the legacy Operator

Reviewer 2: Should we reflect the same statement in the main Readme file of `kubernetes-operator` repository?

Author: You mean the `description` in CSV? 👍  to that. We can actually sneak in the change in https://github.com/keycloak/keycloak/pull/11247 maybe.

Reviewer 2: I mean here:
https://github.com/keycloak/keycloak-operator/blob/main/README.md

Author: Sure. Together with some deprecation note.
